nf_core/subworkflows/install.py

```diff
@@ -27,11 +24,16 @@ def __init__(
         remote_url=None,
         branch=None,
         no_pull=False,
+        installed_by=False,
     ):
         super().__init__(pipeline_dir, remote_url, branch, no_pull)
         self.force = force
         self.prompt = prompt
         self.sha = sha
+        if installed_by:
+            self.installed_by = installed_by
+        else:
+            self.installed_by = self.component_type
```

Comment:
again, for subworkflow I think it should be the subworkflow name, so we can correctly handle modules installed by multiple subworkflows